warehouse/oidc/forms/activestate.py

```diff
@@ -0,0 +1,245 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from typing import TypedDict
+
+import requests
+import sentry_sdk
+import wtforms
+
+from warehouse import forms
+from warehouse.i18n import localize as _
+from warehouse.utils.project import PROJECT_NAME_RE
+
+_VALID_ORG_URL_NAME = re.compile(r"^[a-zA-Z0-9-]{3,40}$")
+_VALID_PROJECT_NAME = re.compile(r"^[.a-zA-Z0-9-]{3,40}$")
```

Comment:
Just sanity checking these... `---` or `-foo-` is a valid "org url" name? `...` or `-.-` is a valid project name?

Reply:
They are not.  I've missed a few validations here.  Fixing and adding a couple more test cases.